frappe/database/database.py

```diff
@@ -990,16 +990,21 @@ def bulk_insert(self, doctype, fields, values, ignore_duplicates=False):
 		insert_list = []
 		fields = ", ".join(["`"+field+"`" for field in fields])
 
-		for idx, value in enumerate(values):
+		if len(values) > 10000:
```

Comment:
Bulk insert is used to insert data in chunks of 10000 and then insert them directly into db.
So adding the `len()` condition would defeat that purpose.